builtin/providers/aws/resource_aws_api_gateway_api_key.go

```diff
@@ -0,0 +1,225 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsApiGatewayApiKey() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsApiGatewayApiKeyCreate,
+		Read:   resourceAwsApiGatewayApiKeyRead,
+		Update: resourceAwsApiGatewayApiKeyUpdate,
+		Delete: resourceAwsApiGatewayApiKeyDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"enabled": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
```

Comment:
Do you think `Default: true` would make more sense? The question is _"Are people likely going to have more enabled API Keys than disabled ones?"_ Also do people expect to explicitly define `enabled = true` in the DSL to make things work? I'd personally say they do not, but I'm open for suggestions/other opinions.

I also understand (now playing devil's advocate with myself) that you were trying to mimic the API's behaviour here.